### ISSUE TITLE
status of bats tests

### ISSUE THREAD

**Issue by FlorianHeigl** [docopt/docopts#15]
Hi Sylvain,

I've modified the shebangs in my clone to /usr/bin/env, and installed a bash4 (i cannot neither confirm or deny if asked if my standard shell is ksh93)

I then also did that to the helpers in the tests/ directory that are there for bats.
Currently I still get a few failures and you could help me a lot by letting me know if they're only on my end.

```
utopia:tests floh$ bats .
 ✗ docopt_auto_parse testing internal behavior
   (in test file ./docopts-auto.bats, line 39)
     `[[ $status == 1 ]]' failed
   ./tmp-docopt_auto_parse.sh
   panic: "usage:" (case-insensitive) not found.
   
   goroutine 1 [running]:
   main.main()
   	/Users/floh/golang/src/github.com/Sylvain303/docopts/docopts.go:423 +0xd73
   status=2
 ✗ docopt_auto_parse functionnal testing
   (in test file ./docopts-auto.bats, line 68)
     `[[ "$output" == prout ]]' failed
   ./tmp-docopt_auto_parse.sh
 ✓ no source
 ✓ global eval
 ✓ docopts error
 ✓ --no-declare
 ✗ docopt_get_help_string
   (in test file ./docopts.bats, line 24)
     `[[ "${lines[0]}" =~ $regexp ]]' failed
 ✓ docopt_get_values
 ✓ docopt_get_eval_array
 ✓ docopt_get_raw_value
 ✓ docopt_print_ARGS

11 tests, 3 failures
```

They seem to occur in standard comparisons. I'll try on a Linux box when I got time again.
If those errors are normal, I'd love to know so I don't hunt a phantom. :)

**Comment by Sylvain303** [comment]
@FlorianHeigl thanks for testing. I merged the alpha release and I need to rewrite some local path too.

If examples are working as expected and non bats test are passing it's already quite good.

I will fix those tests so they will be more easy to read.

Bats unit test suite is for bash so I can't tell result run under ksh. I will move to [bats-core 
fork](https://github.com/bats-core/bats-core) may be it would help.

Could you commit a branch with your changes which makes the tests failing so I could look at the same code?

**Comment by FlorianHeigl** [comment]
Hi, sorry for the misunderstanding, I ran the bats tests under bash4.
I'll keep you posted, I'm trying to get the tests working at all :-)

**Comment by Sylvain303** [comment]
Hi @FlorianHeigl,

I think I've reproduced your environment on travis for this issue:
https://travis-ci.org/Sylvain303/docopts/jobs/394891663

It is probably related to bash 4 path. I needed to fix path for bash in `testee.sh`:

https://github.com/Sylvain303/docopts/commit/673f10527b8768f4e047f81dfac39c9b80d839e1

I've 36 more occurrences of `/bin/bash` mostly in `examples/`.

I think all is related to path of the bash shell installed by `brew install bash`. I made some tests, but it still actually failing.

https://travis-ci.org/Sylvain303/docopts/jobs/394901147

Do you know, if it is possible to install in place of `/bin/bash`?

**Comment by FlorianHeigl** [comment]
I think it'll be far too dangerous, the core of OSX still is using /bin/sh but there's a lot of things that don't stick to it and use bash instead. 
If your homebrew'd break it could be a bit ugly.

I don't think this is a highprio issue, my target platform is Linux mostly... 
But if you want I can go trough everywhere and do the /usr/bin/env thing, it is a good practice and easy.

**Comment by Sylvain303** [comment]
Hi @FlorianHeigl 

I changed shebangs to `#!/usr/bin/env bash` in every scripts, solving #24. I think it also fixes this issue. I also removed usage of `readlink`, that was not used a lot. So the documentation is also fixed about macos limitation on bash 3.2. 

All test are running on Travis [#32](https://travis-ci.org/docopt/docopts/builds/542616027)

GNU/Linux + macos

I didn't yet provide more advanced test without using associative array, which all break on bash3.2

I close this issue, status worksform, the problem should be fixed.
